### PR TITLE
#474: Add Page Search

### PR DESCRIPTION
**Pull request by q-src**
Addresses #474 
Todo:
- [x] mobile search in dedicated menu

### CODE REVIEW

File: src/assets/js/search.js [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+   *
+   * Result format:
+   * {
+   *  "{group-name}": SearchResult[]
+   * }
+   *
+   * @param filter The search term
+   * @returns {{}}
+   */
+  Search.prototype.search = function(filter) {
```
Comment (almeidap) [inline src/assets/js/search.js]: Please, use a more explicit function signature here, e.g. `Search.prototype.filter= function(pattern) {}` where `pattern` could be used with a RegExp to match elements.

**Comment by almeidap** [comment]
Unfortunately, it looks like you missed Typeahead.js, which is already a dependency of the project, for filtering, highlighting and providing suggestions.

Please, take a look at https://github.com/admin-ch/typeahead.js/tree/0.11-templating for improved Typeahead.js view rendering.

**Comment by q-src** [comment]
@almeidap I've updated the implementation so it uses the typeahead fork.

**Comment by almeidap** [comment]
Nice work! With Typeahead, you get everything for free, including accessibility (mainly keyboard navigation).

**Pull request merged**